dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/statelisteners/ElasticListener.java

```diff
@@ -337,6 +335,9 @@ private static Entry removeIrrelevantProperties(final Entry entry) {
         detachedEntry.setCheckerWorkflow(entry.getCheckerWorkflow());
         Set<Version> detachedVersions = cloneWorkflowVersion(entry.getWorkflowVersions());
         detachedEntry.setWorkflowVersions(detachedVersions);
+        // This is some weird hack to always set the topic (which is either automatic or manual) into the ES topicAutomatic property for search table
```

Comment:
Another option could be to have `Entry` have a `@Transient` property (from the DB POV) with the "right" topic selected and for the front-end+ES use that all the time.

Reply:
generated topic column refers to https://www.postgresql.org/docs/current/ddl-generated-columns.html
the transient property and the generated topic column was considered during the draft PR but ultimately wasn't used. adds an extra property to the JSON object that isn't used half of the time (the my-workflows and my-tools page needs to show both topics regardless).
It is my hope that we will someday have a custom Elasticsearch object/class that differs from Entry.java (we currently manually add several properties). that should remove all the unneccessary properties and add the necessary ones. "topic" can be added then.